Please fix the following linting errors:
```
************* Module ceph_bootstrap.terminal_utils
ceph_bootstrap/terminal_utils.py:23:0: R0205: Class 'PrettyPrinter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
ceph_bootstrap/terminal_utils.py:30:4: R0205: Class 'Colors' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
ceph_bootstrap/terminal_utils.py:23:0: R0904: Too many public methods (28/20) (too-many-public-methods)
ceph_bootstrap/terminal_utils.py:265:0: R0205: Class 'PrettyFormat' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
```

--- a/.pylintrc
+++ b/.pylintrc
@@ -102,6 +102,7 @@ disable=print-statement,
         too-many-statements,
         redefined-builtin,
         no-self-use,
+        too-many-public-methods,
 
 enable=c-extension-no-member
 

--- a/ceph_bootstrap/config_shell.py
+++ b/ceph_bootstrap/config_shell.py
@@ -10,6 +10,7 @@ from configshell_fb.shell import locatedExpr
 
 from .core import CephNodeManager, SshKeyManager
 from .salt_utils import PillarManager
+from .terminal_utils import PrettyPrinter as PP
 
 
 logger = logging.getLogger(__name__)
@@ -167,6 +168,7 @@ class SSHGroupHandler(OptionHandler):
         private_key, public_key = SshKeyManager.generate_key_pair()
         PillarManager.set('ceph-salt:ssh:private_key', private_key)
         PillarManager.set('ceph-salt:ssh:public_key', public_key)
+        PP.pl_green('Key pair generated.')
 
     def value(self):
         stored_priv_key = PillarManager.get('ceph-salt:ssh:private_key')
@@ -219,9 +221,11 @@ class TimeServerGroupHandler(OptionHandler):
 
     def enable(self):
         PillarManager.set('ceph-salt:time_server:enabled', True)
+        PP.pl_green('Enabled.')
 
     def disable(self):
         PillarManager.set('ceph-salt:time_server:enabled', False)
+        PP.pl_green('Disabled.')
 
     def value(self):
         val = PillarManager.get('ceph-salt:time_server:enabled')
@@ -513,6 +517,7 @@ class OptionNode(configshell.ConfigNode):
             self.option_dict['handler'].reset()
         else:
             self.value = None
+        PP.pl_green('Value reset.')
 
     def _read_only(self):
         if 'handler' in self.option_dict:
@@ -534,6 +539,7 @@ class ValueOptionNode(OptionNode):
             self.option_dict['handler'].save(value)
         else:
             self.value = value
+        PP.pl_green('Value set.')
 
     def ui_complete_set(self, parameters, text, current_param):
         matching = []
@@ -560,12 +566,14 @@ class FlagOptionNode(OptionNode):
         Enables the option
         '''
         self._set_option_value(True)
+        PP.pl_green('Enabled.')
 
     def ui_command_disable(self):
         '''
         Disables the option
         '''
         self._set_option_value(False)
+        PP.pl_green('Disabled.')
 
 
 class ListElementNode(configshell.ConfigNode):
@@ -593,12 +601,18 @@ class ListOptionNode(OptionNode):
             self.value.append(value)
             self.option_dict['handler'].save(self.value)
             ListElementNode(value, self)
+            PP.pl_green('Value added.')
+        else:
+            PP.pl_red('Value already exists.')
 
     def ui_command_remove(self, value):
         if value in self.value:
             self.value.remove(value)
             self.option_dict['handler'].save(self.value)
             self.remove_child(self.get_child(value))
+            PP.pl_green('Value removed.')
+        else:
+            PP.pl_red('Value not found.')
 
 
 class MinionOptionNode(configshell.ConfigNode):
@@ -631,11 +645,19 @@ class MinionsOptionNode(OptionNode):
 
     def ui_command_add(self, minion_id):
         matching = fnmatch.filter(self.option_dict['handler'].possible_values(), minion_id)
+        counter = 0
         for match in matching:
             if match not in self.value:
                 self.value.append(match)
                 self.option_dict['handler'].save(self.value)
                 MinionOptionNode(match, self.option_dict['handler'].children_handler(match), self)
+                counter += 1
+        if counter == 1:
+            PP.pl_green('1 minion added.')
+        elif counter > 1:
+            PP.pl_green('{} minions added.'.format(counter))
+        else:
+            PP.pl_red('No minions matched "{}".'.format(minion_id))
 
     def ui_command_rm(self, minion_id):
         matching = fnmatch.filter(self.value, minion_id)
@@ -643,6 +665,13 @@ class MinionsOptionNode(OptionNode):
             self.value.remove(match)
             self.option_dict['handler'].save(self.value)
             self.remove_child(self.get_child(match))
+        counter = len(matching)
+        if counter == 1:
+            PP.pl_green('1 minion removed.')
+        elif counter > 1:
+            PP.pl_green('{} minions removed.'.format(counter))
+        else:
+            PP.pl_red('No minions matched "{}".'.format(minion_id))
 
     # pylint: disable=unused-argument
     def ui_complete_add(self, parameters, text, current_param):
@@ -721,7 +750,7 @@ def run_config_shell():
             break
         except Exception as ex:  # pylint: disable=broad-except
             logger.exception(ex)
-            print("An error occurred: {}".format(ex))
+            PP.pl_red(ex)
 
 
 def run_config_cmdline(cmdline):
@@ -730,7 +759,6 @@ def run_config_cmdline(cmdline):
     try:
         logger.info("running command: %s", cmdline)
         shell.run_cmdline(cmdline)
-        print("OK")
     except Exception as ex:  # pylint: disable=broad-except
         logger.exception(ex)
-        print("An error occurred: {}".format(ex))
+        PP.pl_red(ex)

--- a/ceph_bootstrap/terminal_utils.py
+++ b/ceph_bootstrap/terminal_utils.py
@@ -1,0 +1,274 @@
+import logging
+import pprint
+import sys
+
+
+logger = logging.getLogger(__name__)
+
+
+def check_terminal_utf8_support():
+    """
+    Checks whether the terminal supports UTF-8 glyphs.
+    """
+    symbol = u"\u23F3"
+    if sys.stdout.encoding is None:
+        return False
+    try:
+        symbol.encode(sys.stdout.encoding)
+        return True
+    except UnicodeEncodeError:
+        return False
+
+
+class PrettyPrinter:
+    """
+    Helper class to pretty print
+    """
+
+    _PP = pprint.PrettyPrinter(indent=1)
+
+    class Colors:
+        """
+        Color enum
+        """
+        HEADER = '\x1B[95m'
+        BOLD = '\x1B[1m'
+        UNDERLINE = '\x1B[4m'
+        RED = '\x1B[38;5;196m'
+        GREEN = '\x1B[38;5;83m'
+        DARK_GREEN = '\x1B[38;5;34m'
+        YELLOW = '\x1B[38;5;226m'
+        DARK_YELLOW = '\x1B[38;5;178m'
+        BLUE = '\x1B[38;5;33m'
+        MAGENTA = '\x1B[38;5;198m'
+        CYAN = '\x1B[38;5;43m'
+        ORANGE = '\x1B[38;5;214m'
+        PURPLE = '\x1B[38;5;134m'
+        GREY = '\x1B[38;5;245m'
+        LIGHT_YELLOW = '\x1B[38;5;228m'
+        LIGTH_PURPLE = '\x1B[38;5;225m'
+        ENDC = '\x1B[0m'
+
+    @classmethod
+    def _format(cls, color, text):
+        """
+        Generic pretty print string formatter
+        """
+        return u"{}{}{}".format(color, text, cls.Colors.ENDC)
+
+    @classmethod
+    def header(cls, text):
+        """
+        Formats text as header
+        """
+        return cls._format(cls.Colors.HEADER, text)
+
+    @classmethod
+    def bold(cls, text):
+        """
+        Formats text as bold
+        """
+        return cls._format(PrettyPrinter.Colors.BOLD, text)
+
+    @classmethod
+    def blue(cls, text):
+        """
+        Formats text as blue
+        """
+        return cls._format(cls.Colors.BLUE, text)
+
+    @classmethod
+    def grey(cls, text):
+        """
+        Formats text as grey
+        """
+        return cls._format(PrettyPrinter.Colors.GREY, text)
+
+    @staticmethod
+    def light_purple(text):
+        """
+        Formats text as light_purple
+        """
+        return PrettyPrinter._format(PrettyPrinter.Colors.LIGTH_PURPLE, text)
+
+    @staticmethod
+    def green(text):
+        """
+        Formats text as green
+        """
+        return PrettyPrinter._format(PrettyPrinter.Colors.GREEN, text)
+
+    @staticmethod
+    def dark_green(text):
+        """
+        Formats text as dark_green
+        """
+        return PrettyPrinter._format(PrettyPrinter.Colors.DARK_GREEN, text)
+
+    @staticmethod
+    def yellow(text):
+        """
+        Formats text as yellow
+        """
+        return PrettyPrinter._format(PrettyPrinter.Colors.YELLOW, text)
+
+    @staticmethod
+    def dark_yellow(text):
+        """
+        Formats text as dark_yellow
+        """
+        return PrettyPrinter._format(PrettyPrinter.Colors.DARK_YELLOW, text)
+
+    @staticmethod
+    def red(text):
+        """
+        Formats text as red
+        """
+        return PrettyPrinter._format(PrettyPrinter.Colors.RED, text)
+
+    @staticmethod
+    def orange(text):
+        """
+        Formats text as orange
+        """
+        return PrettyPrinter._format(PrettyPrinter.Colors.ORANGE, text)
+
+    @staticmethod
+    def cyan(text):
+        """
+        Formats text as cyan
+        """
+        return PrettyPrinter._format(PrettyPrinter.Colors.CYAN, text)
+
+    @staticmethod
+    def magenta(text):
+        """
+        Formats text as magenta
+        """
+        return PrettyPrinter._format(PrettyPrinter.Colors.MAGENTA, text)
+
+    @staticmethod
+    def purple(text):
+        """
+        Formats text as purple
+        """
+        return PrettyPrinter._format(PrettyPrinter.Colors.PURPLE, text)
+
+    @staticmethod
+    def info(text):
+        """
+        Formats text as info
+        """
+        return PrettyPrinter._format(PrettyPrinter.Colors.LIGHT_YELLOW, text)
+
+    @staticmethod
+    def p_header(text):
+        """
+        Prints text formatted as header
+        """
+        print(PrettyPrinter.header(text))
+
+    @staticmethod
+    def p_bold(text):
+        """
+        Prints text formatted as bold
+        """
+        sys.stdout.write(PrettyPrinter.bold(text))
+        sys.stdout.flush()
+
+    @staticmethod
+    def pl_bold(text):
+        """
+        Prints text formatted as bold with newline in the end
+        """
+        sys.stdout.write(u"{}\n".format(PrettyPrinter.bold(text)))
+        sys.stdout.flush()
+
+    @classmethod
+    def print(cls, text):
+        """
+        Prints text as is
+        """
+        sys.stdout.write(text)
+        sys.stdout.flush()
+
+    @classmethod
+    def println(cls, text=None):
+        """
+        Prints text as is with newline in the end
+        """
+        if text:
+            sys.stdout.write(u"{}\n".format(text))
+            sys.stdout.flush()
+        else:
+            sys.stdout.write(u"\n")
+            sys.stdout.flush()
+
+    @classmethod
+    def p_blue(cls, text):
+        """
+        Prints text formatted as blue
+        """
+        cls.print(cls.blue(text))
+
+    @classmethod
+    def pl_blue(cls, text):
+        """
+        Prints text formatted as blue
+        """
+        cls.println(cls.blue(text))
+
+    @classmethod
+    def p_green(cls, text):
+        """
+        Prints text formatted as green
+        """
+        cls.print(cls.green(text))
+
+    @classmethod
+    def pl_green(cls, text):
+        """
+        Prints text formatted as green
+        """
+        cls.println(cls.green(text))
+
+    @staticmethod
+    def p_red(text):
+        """
+        Prints text formatted as red
+        """
+        print(PrettyPrinter.red(text))
+
+    @classmethod
+    def pl_red(cls, text):
+        """
+        Prints text formatted as green
+        """
+        cls.println(cls.red(text))
+
+    @staticmethod
+    def flush():
+        """
+        Flush stdout
+        """
+        sys.stderr.flush()
+        sys.stdout.flush()
+
+    @staticmethod
+    def format_dict(dict_obj):
+        """
+        Formats a dict structure using pprint formatter
+        """
+        return PrettyPrinter._PP.pformat(dict_obj)
+
+
+class PrettyFormat:
+
+    OK = PrettyPrinter.green(PrettyPrinter.bold(u"\u2713")) \
+        if check_terminal_utf8_support() else PrettyPrinter.green("OK")
+
+    FAIL = PrettyPrinter.red(u"\u274C") \
+        if check_terminal_utf8_support() else PrettyPrinter.red("Fail")
+
+    WAITING = PrettyPrinter.orange(u"\u23F3") \
+        if check_terminal_utf8_support() else PrettyPrinter.orange("Running")

